include/dfu/suit_dfu_fetch_source.h

```diff
@@ -16,45 +29,39 @@
 extern "C" {
 #endif
 
-/*
- * The fetch source manager allows for registration of multiple data sources to be used by
- * the SUIT processor.
- * The SUIT proccessor iterates over the registered fetch sources, calling respective
- * suit_dfu_fetch_source_request_fn.
- */
-
 /**
  * @brief Fetch source callback function.
- *        Fetch source callbacks are registered via suit_dfu_fetch_source_register.
- *        Multiple such callbacks can be registered. The callbacks are called from inside
- *        the SUIT processor until one of them succeeds.
- *        Iteration will be broken if one of registered fetch sources starts to call
- *        @ref suit_dfu_fetch_source_write_fetched_data or @ref suit_dfu_fetch_source_seek.
  *
- *        The role of the callback is to fetch data from the provided URI and pass it to SUIT
- *        via @ref suit_dfu_fetch_source_write_fetched_data - the data does not have to be passed
- *        all at once, @ref suit_dfu_fetch_source_write_fetched_data can be called multiple times
- *        as new chunks of data arrive.
- *        If the data needs to be written at some offset (e.g. if the chunks arrive out of order)
- *        @ref suit_dfu_fetch_source_seek can be used to move the write pointer to the given offset.
+ * @details
+ * Fetch source callbacks are registered via suit_dfu_fetch_source_register.
+ * Multiple such callbacks can be registered. The callbacks are called from inside
+ * the SUIT processor until one of them succeeds.
+ * Iteration will be broken if one of registered fetch sources starts to call
+ * @ref suit_dfu_fetch_source_write_fetched_data or @ref suit_dfu_fetch_source_seek.
```

Comment:
What "iteration will be broken" means? From the context I assume you meant: "A callback function reports its success by calling one of @ref suit_dfu...". If I'm correct, please rephrase for clarification.

Reply:
Or maybe you meant "If the callback function is to return an error, MUST NOT call any of @ref suit_dfu_fetch... functions"
If this guess is correct, I would move this requirement after the next paragraph describing when these functions are used by the callback thread.

Reply:
To my understanding the thing works that way that it calls callbacks as long as they return error and do not attempt to write data through `suit_dfu_fetch_source_write_fetched_data` or `suit_dfu_fetch_source_seek`.

If the callback succeeds - the operations is assumed to be done.
If the callback fails after writing data - processing of the SUIT directive fails, no other fetch source is checked and the SUIT processor interprets the result code.

Reply:
This sentence should be rephrased to what you explained in the comment: something around "If the callback calls `suit_dfu_fetch_...` or `suit_dfu_fetch...` and returns an error, it is a SUIT directive fault. The SUIT processor handles it by ..." (aborting the processing?).

Reply:
Rephrased.